### ISSUE TITLE
Support for nullish coalescing and optional chaining syntax like esint

### ISSUE THREAD

**Issue by diegoazh** [vuejs/vetur#1613]
- [x] I have searched through existing issues
- [x] I have read through [docs](https://vuejs.github.io/vetur)
- [x] I have read [FAQ](https://github.com/vuejs/vetur/blob/master/docs/FAQ.md)

## Info

- Platform:  Linux
- Vetur version: 0.22.6
- VS Code version: 1.41.1

## Problem

When use optional chaining operator vertur shows a syntax error highlight with the followin message:

`Expression expected.Vetur(1109)`

## Reproducible Case

write a function like this

```javascript
    isImage(answer) {
      return !!answer?.media?.image;
    },
```

**Comment by Niekvdm** [comment]
This was introduced in TypeScript 3.7. As of now Vetur uses 3.6.x

**Comment by diegoazh** [comment]
> This was introduced in TypeScript 3.7. As of now Vetur uses 3.6.x

Ok, in that case I think that an upgrade of TypeScript is a good idea in order to get the same behavior on vscode.

**Comment by kieraneglin** [comment]
We're using this via Babel transformers, not TypeScript. Will the upgrade to 3.7 resolve this in our case as well?

**Comment by 3nuc** [comment]
What you're describing is already possible.
Set `vetur.useWorkspaceDependencies: true` in settings.json and make sure you have TS 3.7 installed in node_modules. You can probably even use the new TS 3.8 beta but I haven't tried yet.

Nullish coalescing and optional chaining won't work in templates though - only in <script> tag.

**Comment by octref** [comment]
Should already be fixed in 0.23.0.
